src/providers/python_controller.js

```diff
@@ -51,26 +46,23 @@ export default class Python_controller {
 		this.pathToRequirements = pathToRequirements
 		this.options = options
 	}
-	prepareEnvironment()
-	{
+	prepareEnvironment() {
 		if(!this.realEnvironment) {
 			this.pythonEnvDir = path.join(path.sep,"tmp","exhort_env_js")
-			execSync(`${handleSpacesInPath(this.pathToPythonBin)} -m venv ${handleSpacesInPath(this.pythonEnvDir)} `, err => {
-				if (err) {
-					throw new Error('failed creating virtual python environment - ' + err.message)
-				}
-			})
-			if(this.pathToPythonBin.includes("python3"))
-			{
+			try {
+				invokeCommand(this.pathToPythonBin, ['-m', 'venv', this.pythonEnvDir])
+			} catch (error) {
+				throw new Error('failed creating virtual python environment - ' + error.message)
```

Comment:
Here too. Also I think it's better to be consistent in the error messages `fail` vs `failed` and in Javascript is a best practice to write error messages as a full sentence, starting with capital letter.
```suggestion
				throw new Error('Failed creating virtual python environment', { cause: error });
```